src/lib/Imp.hs

```diff
@@ -1310,7 +1308,7 @@ computeElemCount (EmptyAbs Empty) =
   return $ IdxRepVal 1
 computeElemCount idxNest' = do
   let (idxList, idxNest) = indexStructureSplit idxNest'
-  sizes <- forM idxList indexSetSizeFromSimp
+  sizes <- forM idxList indexSetSize
```

Comment:
How does that work? `indexSetSize` doesn't look into the top level bindings and I thought that at this point the Ix dicts should be represented by names?

Reply:
`indexSetSize` calls `applyIxMethod`, which does look at the top level bindings in the `ExplicitMethods` case.